### PR TITLE
Updated WSL version 2.1.5.0

### PR DESCRIPTION
**Pull request by craigloewen-msft**
Checklist for Pull Requests
- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?
No

Manifests
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] This PR only modifies one (1) manifest
- [X] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [X] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/151701)

### CODE REVIEW

File: manifests/m/Microsoft/WSL/2.1.5.0/Microsoft.WindowsSubsystemforLinux.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Author: Microsoft
+PackageName: Windows Subsystem for Linux
+PackageUrl: https://learn.microsoft.com/windows/wsl/about
+License: Microsoft Software License
+LicenseUrl: https://www.microsoft.com/servicesagreement
+Copyright: Copyright (c) Microsoft Corporation. All rights reserved.
+ShortDescription: Windows Subsystem for Linux (WSL) is a feature of Windows that allows you to run a Linux environment on your Windows machine, without the need for a separate virtual machine or dual booting.
+Description: Windows Subsystem for Linux (WSL) is a feature of Windows that allows you to run a Linux environment on your Windows machine, without the need for a separate virtual machine or dual booting. WSL is designed to provide a seamless and productive experience for developers who want to use both Windows and Linux at the same time.
+Moniker: wsl
+ReleaseNotesUrl: https://github.com/microsoft/WSL/releases/tag/2.2.2
```
Comment (OneBlue) [inline manifests/m/Microsoft/WSL/2.1.5.0/Microsoft.WindowsSubsystemforLinux.locale.en-US.yaml:20]: Shouldn't this be https://github.com/microsoft/WSL/releases/tag/2.1.5 ?

**Review by OneBlue — CHANGES_REQUESTED** [review]

**Review by OneBlue — APPROVED** [review]

**Comment by ryfu-msft** [comment]
I think this is failing because the filename needs to match the package identifier that you are using `Microsoft.WSL.yaml`

**Review by denelon — DISMISSED** [review]

File: manifests/m/Microsoft/WSL/2.1.5.0/Microsoft.WSL.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+InstallerType: wix
+Installers:
+- InstallerUrl: https://github.com/microsoft/WSL/releases/download/2.1.5/wsl.2.1.5.0.x64.msi
+  Architecture: x64
+  InstallerSha256: C2B94E81AB4BA2780016CB28A7C07C87B570F65D26DC7D3DC91A1016D69028AF
+  ProductCode: '{877F46EF-614F-4B05-A09D-E15E5B424710}'
+- InstallerUrl: https://github.com/microsoft/WSL/releases/download/2.1.5/wsl.2.1.5.0.ARM64.msi
+  Architecture: arm64
+  InstallerSha256: FFCCE6428255D3B9FC9C4BD1D584E1282936879287F52988F0D4056BF8377914
+  ProductCode: '{350E3AAF-271D-4B51-9D76-9AAD330A8536}'
```
Comment (denelon) [inline manifests/m/Microsoft/WSL/2.1.5.0/Microsoft.WSL.installer.yaml:16]:
```suggestion
  ProductCode: '{350E3AAF-271D-4B51-9D76-9AAD330A8536}'
Dependencies:
  WindowsFeatures:
  - Microsoft-Windows-Subsystem-Linux
  - VirtualMachinePlatform
```

**Comment by denelon** [comment]
@stephengillie I think what is happening is the DISM command isn't working in the validation VM's version of Windows. There are two features here and I'm guessing one is for older versions of Windows and one is for newer versions of Windows. We might need to do a minOSVersion to specify one for the older version(s) of Windows and the other for newer version(s) of Windows.

**Comment by denelon** [comment]
@craigloewen-msft do you know which Windows version is the minimum for both Windows Optional Features? We might be able to tweak the manifest so it can handle the right installer behavior based on the minOSVersion?

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 8:23:59 AM: 
> 2024-05-02 08:23:58.779 [CORE] Failed to enable Windows Feature [Microsoft-Windows-Subsystem-Linux] with exit code: 14081

(Automated response - build 880.)

**Review by denelon — APPROVED** [review]

**Pull request merged**